internal/collector/postgres_metrics.go

```diff
@@ -180,31 +181,65 @@ func EnablePostgresMetrics(ctx context.Context, inCluster *v1beta1.PostgresClust
 			Exporters: []ComponentID{Prometheus},
 		}
 
-		// Add custom queries if they are defined in the spec
+		// Add custom queries and per-db metrics if they are defined in the spec
 		if inCluster.Spec.Instrumentation != nil &&
-			inCluster.Spec.Instrumentation.Metrics != nil &&
-			inCluster.Spec.Instrumentation.Metrics.CustomQueries != nil &&
-			inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add != nil {
-
-			for _, querySet := range inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add {
-				// Create a receiver for the query set
-				receiverName := "sqlquery/" + querySet.Name
-				config.Receivers[receiverName] = map[string]any{
-					"driver": "postgres",
-					"datasource": fmt.Sprintf(
-						`host=localhost dbname=postgres port=5432 user=%s password=${env:PGPASSWORD}`,
-						MonitoringUser),
-					"collection_interval": querySet.CollectionInterval,
-					// Give Postgres time to finish setup.
-					"initial_delay": "15s",
-					"queries": "${file:/etc/otel-collector/" +
-						querySet.Name + "/" + querySet.Queries.Key + "}",
+			inCluster.Spec.Instrumentation.Metrics != nil {
+
+			if inCluster.Spec.Instrumentation.Metrics.CustomQueries != nil &&
+				inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add != nil {
+
+				for _, querySet := range inCluster.Spec.Instrumentation.Metrics.CustomQueries.Add {
+					// Create a receiver for the query set
+
+					dbs := []string{"postgres"}
+					if len(querySet.Databases) != 0 {
+						dbs = querySet.Databases
+					}
```

Comment:
Definitely need to make a note in the documentation that if the user provides any databases for a custom query set that the default "postgres" database will not be included (unless they include it themselves).